frontend/src/views/SecretOverviewPage/SecretOverviewPage.tsx

```diff
@@ -236,6 +242,64 @@ export const SecretOverviewPage = () => {
     }
   };
 
+  const handleFolderUpdate = async (newFolderName: string) => {
+    const { name: oldFolderName } = popUp.updateFolder.data as TSecretFolder;
+    let processedUpdates = 0;
+    let totalFolderMatchCount = 0;
+
+    await Promise.allSettled(
+      userAvailableEnvs.map(async (env) => {
+        if (
+          permission.cannot(
+            ProjectPermissionActions.Edit,
+            subject(ProjectPermissionSub.Secrets, { environment: env.slug, secretPath })
+          )
+        ) {
+          return;
+        }
+
+        const folder = getFolderByNameAndEnv(oldFolderName, env.slug);
+        if (folder) {
+          totalFolderMatchCount += 1;
+
+          await updateFolder({
+            folderId: folder.id,
+            name: newFolderName,
+            path: secretPath,
+            environment: env.slug,
+            projectId: workspaceId
+          });
+
+          processedUpdates += 1;
+        }
+      })
+    );
+
+    if (totalFolderMatchCount === 0) {
+      createNotification({
+        type: "info",
+        text: "You don't have access to rename selected folder"
+      });
+    } else if (processedUpdates === totalFolderMatchCount) {
+      createNotification({
+        type: "success",
+        text: "Successfully renamed folder across environments"
+      });
+    } else if (processedUpdates > 0) {
+      createNotification({
+        type: "warning",
+        text: "Partially renamed folder across environments"
+      });
```

Comment:
I'm not sure if it's a good idea to allow partial updates. I feel like it would make more sense to have a "bulk update" endpoint like we have with secrets, which would be wrapped in a transaction. That way you can always expect all your data to be updated correctly, or it will throw an error and nothing will be updated

Reply:
Yeah, i agree. All or nothing for rename at the overview level because the user is intending to change it for all of the environments

Reply:
Got this. But to clarify, if a rename can't be made in an environment because a folder with an existing name already exists, should it stop the renaming of all the other folders in other environments?

Reply:
resolved via DM, yes it should error out ^